Cannot build component because of handlebars

**Describe the bug**
When trying to build the project, I get the following Error:
require.extensions is not supported by webpack. Use a loader instead.
 @ ./PolyLookup/components/PolyLookupControl.tsx 173:35-56
 @ ./PolyLookup/index.ts 154:39-80

WARNING in ./node_modules/handlebars/lib/index.js 22:2-20
require.extensions is not supported by webpack. Use a loader instead.
 @ ./PolyLookup/components/PolyLookupControl.tsx 173:35-56
 @ ./PolyLookup/index.ts 154:39-80

ERROR in ./node_modules/handlebars/lib/index.js 15:11-24
Module not found: Error: Can't resolve 'fs' in 'C:\Sources\CustomerProjects\DCE.PCF\PolyLookupComponent\node_modules\handlebars\lib'
resolve 'fs' in 'C:\Sources\CustomerProjects\DCE.PCF\PolyLookupComponent\node_modules\handlebars\lib'
  Parsed request is a module
  using description file: C:\Sources\CustomerProjects\DCE.PCF\PolyLookupComponent\node_modules\handlebars\package.json (relative path: ./lib)
    resolve as module
      C:\Sources\CustomerProjects\DCE.PCF\PolyLookupComponent\node_modules\handlebars\lib\node_modules doesn't exist or is not a directory

**Steps to Reproduce**
1. Clone The Repostory
2. npm install
3. npm run build

Since there is a //TODO: fix this hack at the line that causes this build error (Line 21 PolyLookupControl.tsx, I assume you are already aware of this problem and your fix apparentyl doesnt work everywhere. Any Idea how I can make it work? I am on Windows, vscode, fresh clone.
node --version = v16.17.0

--- a/PolyLookupComponent/PolyLookup/components/PolyLookupControl.tsx
+++ b/PolyLookupComponent/PolyLookup/components/PolyLookupControl.tsx
@@ -18,14 +18,10 @@ import {
   useSelectedItems,
   useSuggestions,
 } from "../services/DataverseService";
-
-//TODO: fix this hack
-// edit node_modules\handlebars\package.json and
-// change "browser" to "browser": "./dist/cjs/handlebars.js",
-import MainHandlebars from "handlebars";
-import * as RuntimeHandlebars from "handlebars/runtime";
 import { SuggestionInfo } from "./SuggestionInfo";
-const Handlebars = Object.assign(MainHandlebars, RuntimeHandlebars);
+
+// TODO: fix this import in handlebars next version
+import Handlebars from "handlebars/lib/handlebars";
 
 const queryClient = new QueryClient();
 const parser = new DOMParser();
@@ -301,7 +297,7 @@ const Body = ({
             borderRadius: 1,
             "&:after": {
               backgroundColor: "transparent",
-              borderColor: isFocused ? "#000" : "transparent",
+              borderColor: isFocused ? "#666" : "transparent",
               borderWidth: 1,
               borderRadius: 1,
             },

--- a/PolyLookupComponent/PolyLookup/index.ts
+++ b/PolyLookupComponent/PolyLookup/index.ts
@@ -36,13 +36,19 @@ export class PolyLookup implements ComponentFramework.ReactControl<IInputs, IOut
   public updateView(context: IExtendedContext): React.ReactElement {
     this.context = context;
 
+    let clientUrl = "";
+
+    try {
+      clientUrl = context.page.getClientUrl();
+    } catch {}
+
     const props: PolyLookupProps = {
       currentTable: context.page.entityTypeName,
       currentRecordId: context.page.entityId,
       relationshipName: context.parameters.relationship.raw ?? "",
       relationshipType:
         RelationshipTypeEnum[context.parameters.relationshipType.raw as keyof typeof RelationshipTypeEnum],
-      clientUrl: context.page.getClientUrl(),
+      clientUrl: clientUrl,
       lookupView: context.parameters.lookupView.raw ?? undefined,
       itemLimit: context.parameters.itemLimit.raw ?? undefined,
       pageSize: context.userSettings.pagingLimit ?? undefined,

--- a/PolyLookupComponent/PolyLookup/types/handlebars.ts
+++ b/PolyLookupComponent/PolyLookup/types/handlebars.ts
@@ -1,0 +1,5 @@
+declare module "handlebars/lib/handlebars" {
+  // Re-export the types from the existing handlebars module
+  import Handlebars = require("handlebars");
+  export default Handlebars;
+}

